src/sentry/snuba/query_subscription_consumer.py

```diff
@@ -46,22 +216,81 @@ class InvalidSchemaError(InvalidMessageError):
     pass
 
 
+class QuerySubscriptionStrategyFactory(ProcessingStrategyFactory[KafkaPayload]):
+    def __init__(self, topic: str):
+        self.topic = topic
+
+    def create_with_partitions(
+        self,
+        commit: Commit,
+        partitions: Mapping[Partition, int],
+    ) -> ProcessingStrategy[KafkaPayload]:
+        def process_message(message: Message[KafkaPayload]) -> None:
+            with sentry_sdk.start_transaction(
+                op="handle_message",
+                name="query_subscription_consumer_process_message",
+                sampled=random() <= options.get("subscriptions-query.sample-rate"),
+            ), metrics.timer("snuba_query_subscriber.handle_message"):
+                value: Any = message.value
```

Comment:
I probably need to fix this up, was getting irritated with typing.

Reply:
Ah yeah, this is a bit confusing since there are 2 types of message values in the library. 

For now you can do something like:
```
value = message.value
assert isinstance(value, BrokerValue)
offset = value.offset
partition = value.partition.index
```